### PR TITLE
feat/preview-comply-with-photo-resolution-when-video-is-disabled

### PR DESCRIPTION
**Pull request by ismaelsousa**
## What

This PR handles the Android preview based on the format and camera configuration, using the proper resolution if the video output is disabled.

### Use case: I want to see exactly what the final photo will look like in the preview.

To achieve this you can use the selected format and calculate the ratio, then you can render your camera view like this:
```js
const aspectRatio = selectedFormat.photoHeight / selectedFormat.photoWidth; // if sensor is rotated

{
  !!device && (
    <View
      style={{
        width: '100%',
        aspectRatio,
      }}>
      <Camera
        ref={cameraRef}
        style={StyleSheet.absoluteFill}
        device={device}
        isActive={true}
        resizeMode="contain"
        format={selectedFormat}
        photo
      />
    </View>
  );
}
```

## Changes

- This PR sets the preview size based on the camera configuration
- This PR updates the documentation of the preview section

## Tested on

- iPhone 15 Pro, iOS 17.6.1
- Android Moto G52, android 13

### CODE REVIEW

File: package/android/src/main/java/com/mrousavy/camera/core/CameraSession+Configuration.kt [hunk 1]
```diff
@@ -75,9 +75,11 @@ internal fun CameraSession.configureOutputs(configuration: CameraConfiguration)
       }
 
       if (format != null) {
-        // Similar to iOS, Preview will follow video size as it's size (and aspect ratio)
+        // If video is disable but photo is enabled, use photoSize
+        val targetSize = if (videoConfig == null) format.photoSize else format.videoSize
```
Comment (mrousavy) [inline package/android/src/main/java/com/mrousavy/camera/core/CameraSession+Configuration.kt]:
```suggestion
        // Preview will follow video size as it's size & aspect ratio, or photo- if video is disabled.
        val targetSize = if (videoConfig != null) format.videoSize else format.photoSize
```

**Review by mrousavy — CHANGES_REQUESTED** [review]
Please revert all the unrelated changes. Not sure what formatter you guys are rocking but this is messing up the docs.

I also flipped the if to make it more readable.

**Comment by ismaelsousa** [comment]
> Please revert all the unrelated changes. Not sure what formatter you guys are rocking but this is messing up the docs.
> 
> I also flipped the if to make it more readable.

Done, I am using the eslint and prettier.
But I disabled it for the whole project now :)

**Comment by mrousavy** [comment]
it still not reverted

**Comment by ismaelsousa** [comment]
image

Do you mean to remove all changes from the docs?
Here I specify that the preview on Android will follow the photo or video prop

**Comment by mrousavy** [comment]
my bad sorry I just quickly peeked at it and thought it was a formatting change. sorry I need sleep.

File: docs/docs/guides/PREVIEW.mdx [hunk 2]
```diff
@@ -62,7 +62,15 @@ const lowResolutionFormat = useCameraFormat(device, [
 ])
 ```
 
-On Android, the Preview will always be in full HD, or the Preview View's size, whichever is smaller.
+On Android, the Preview will always be in full HD or the Preview View's size, whichever is smaller. If a format is specified, the preview will comply with the format's resolution based on whether the `photo` or `video` is set.  E.g., if `video` is not set, the preview will be in the photo resolution. The preview will be in the video resolution if `video` is set.
```
Comment (mrousavy) [inline docs/docs/guides/PREVIEW.mdx]:
```suggestion
On Android, the Preview will always be in full HD or the Preview View's size, whichever is smaller.
If a format is specified, the preview will try to match the `video`'s resolution and aspect ratio if `video` is enabled, and `photo`'s resolution and aspect ratio if `photo` is enabled.
```

**Comment by ismaelsousa** [comment]
cool :) don't worry, go rest, thank you

**Pull request merged**